### DISCUSSION TITLE
Integracja AppService - API Management

### DISCUSSION THREAD

**Discussion by miqm** [AzureCommunityPL/group#293] [Pomoc] [answered]
Hej,

Mam pytanie do was dot. App Service Web Apps - jest tam opcja integracji z API-Management.

Poza tym, że przy jej włączeniu na API-M pojawi się odpowiedni schemat (z wildcardem lub wg podanej specyfikacji), ustawi backend, itd. -  to czy ma to jeszcze jakieś benefity jeśli chodzi o operacyjność?

**Comment by mgrabarz** [comment, marked as answer]
Jak dla nie to ta opcja, z której nie zdarzyło mi się korzystać, po prostu ustawia API/polityki w APIM z automatu. Poza tym nic więcej nie wnosi. 
Dlaczego nie używałem - wolę sam, świadomie wszystko poustawiać zamiast ufać automatom.

**Comment by piotrgwiazda** [comment]
Chodzi o korzystanie z APIM co do zasady? Czy o auto-integrację?

pt., 4 cze 2021, 10:02 użytkownik Marek Grabarz ***@***.***>
napisał:

**Reply by miqm** [reply]
o opcję autointegracji - czy jeśli mam już samodzielnie zbudowane api i policies na API-M, to czy ta integracja (a w zasadzie ustawienie w webappie ID APIM) coś wnosi extra.

**Comment by piotrgwiazda** [comment]
Jeśli robisz publiczne API dla klientów/partnerów lub w większej
organizacji, to zwykle publikujesz OpenAPI, testujesz kontrakty i robisz
nową wersję zanim zaimplementujesz kod.


pt., 4 cze 2021, 13:01 użytkownik Mikolaj Mackowiak <
***@***.***> napisał:

**Reply by mgrabarz** [reply]
Różne podejścia widziałem, ale rozumiem do czego zmierzasz. 
Najpierw kontrakty, potem APIM i polityki z ewentualnymi [mockami](https://docs.microsoft.com/en-us/azure/api-management/api-management-advanced-policies#mock-response), a dopiero potem kod i deployment do App Service. W takim scenariuszu feature rzeczywiście jest kompletnie nieprzydatny.
